### ISSUE TITLE
iOS - Not Able to Undo Bookmark after Setting It with VoiceOver

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#95]
While using VoiceOver the user is not able to Undo the bookmark because it does not focus to the Undo button or at least the message and goes away to quickly.

**Comment by jfc3-dol** [comment]
VoiceOver still does not allow the user to "Undo" the item they bookmarked using the "Undo" button that shows up with which item you bookmarked. Not sure if the message is going away too quickly that it can't rad and focus on the item.

**Comment by KyleLivingstonDOL** [comment]
Thanks John. ZTI, can we fix this? Or is this an inherent issue with Voice Over? I don't think we can keep the Undo popup for much longer than is on the page. Is there a code we can insert that will direct voice over to the Undo popup immediately?

**Comment by harpreetsdev** [comment]
@jfc3-dol for this issue, the Bookmark button itself converts into "Undo Bookmark" button after the user Bookmarks a certain Step/Topic. You can hear the Voice over announcement to "Unbookmark Step XYZ" after the user Bookmarks a particular section/Topic/Step. So essentially Voice over focus doesn't need to move to the bottom of page Undo a bookmark.

In my opinion the Undo Bookmark button is of less value for a VO/Keyboard user in this case and more apt for a user using a pointing device.

We could however change the aria-description to be a little more Verbose or descriptive for the user if you want but I feel taking the focus to bottom of page could potentially cause same issues as #108 where the focus is expected to come back to top of page once the button is pressed.

Please let me know what you think.

**Comment by jfc3-dol** [comment]
@harpreetsdev 

I will have to give this a look tomorrow since I got a lot of other items to review this week for something that is going live Monday. So I need to check on a few other things before the end of today.

**Comment by harpreetsdev** [comment]
@jfc3-dol
 
So I went ahead and added an "Undo Bookmark" description to the Bookmark button once the VO user bookmarks the section. Since the Bookmark description was pretty verbose and explanatory I left that unchanged.

For the Undo bookmark description it should say - " This 'section' is bookmarked, to undo press Enter" suggesting the Voiceover user to use the same button to remove bookmark.

Please let me know if that suffices or other text you intend to add.

**Comment by jfc3-dol** [comment]
@harpreetsdev 

Should I still test the older version I have or wait until these updates have been pushed from TestFlight?

**Comment by harpreetsdev** [comment]
@jfc3-dol 

These changes should be in the newer builds for Web and Native, we are in the process of creating archives to be sent over with some last minute changes. 

Should be in a few hours.

**Comment by jfc3-dol** [comment]
@harpreetsdev 

Thanks for the update.

I will attempt to get them reviewed today if the update arrives before I leave for the day at 4:30 PM. If not, I will look at them on Monday.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Copy that.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

Can you check why after bookmarking an item, the text is repeated twice, such as "You have bookmarked XYZ. You have bookmarked XYZ. Undo". It's like the item to be bookmarked is entered in the aria-live region or notification section twice before adding the text "Undo". Or maybe it is reading the aria-live region text and then what is in the pop-up to Undo and item. 

**NOTE**: XYZ is the screen being bookmarked.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Sure. l can test this and get back to you.. I am also looking at some other VO issues..

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

When using VocieOver the bookmark button on a screen reads "SCREEN NAME" followed by "is bookmarked. To undo bookmark press enter". I would expect it to tell the user to "double-tap" instead " of "press enter".

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

After bookmarking a screen, VoiceOver read the text shown on the Undo pop-up "You have bookmarked" and whatever screen TITLE you are on. Once the Undo pop-up goes away, VoiceOver reads the same text from the pop-up again. I'm assuming it because the bookmark icon/button has that text. 

If you swipe away and return to the bookmark icon/button, VoiceOver reads "SCREEN NAME" followed by "is bookmarked. To undo bookmark, press enter". I would expect it to tell the user to "double-tap" instead " of "press enter".

**Comment by jfc3-dol** [comment]
@pbhatt17 

Everything seems to be working as expected now, so I'm going to close the issue.
